evaluate.rkt

```diff
@@ -1,8 +1,14 @@
 #lang racket
 
-(require "src/fpcore-checker.rkt" "src/fpcore-interpreter.rkt" "src/fpcore-reader.rkt")
+(require "src/fpcore-checker.rkt" "src/fpcore-interpreter.rkt" "src/fpcore-reader.rkt" "src/multi-command-line.rkt")
 
-(provide evaluate-main)
+(provide evaluate-main evaluate-body make-evaluate-ctx)
+
+; CLI args passed in through context
+(struct evaluate-ctx (in-file out-file check-types? ragged-check? args) #:transparent)
+
+(define (make-evaluate-ctx in-file out-file check-types? ragged-check? args)
+  (evaluate-ctx in-file out-file check-types? ragged-check? args))
```

Comment:
This seems a little strange to me. Why not a procedure with an annoying amount of arguments?

Reply:
I considered this but I remember when we did the egraph-query in Herbie we did something like this and it can make the body of then mains a bit more readable.